Mathlib/GroupTheory/Perm/Fin.lean

```diff
@@ -306,6 +301,132 @@ end Fin
 
 end CycleRange
 
+section cycleIcc
+
+/-! ### `cycleIcc` section
+
+* Define the permutation `cycleIcc i j`, which is the cycle `(i i+1 .... j)` leaving
+`(0 ... i-1)` and `(j+1 ... n-1)` unchanged when `i ≤ j` and returns the dummy value identity when
+`i > j`. In other words, it rotates elements in `[i, j]` one step to the right.
+-/
+
+namespace Fin
+
+local instance {n : ℕ} {i : Fin n} : NeZero (n - i) := NeZero.of_pos (by omega)
+
+variable {n : ℕ} {i j k : Fin n}
+
+/-- `cycleIcc i j` is the cycle `(i i+1 .... j)` leaving `(0 ... i-1)` and `(j+1 ... n-1)`
+unchanged when `i ≤ j` and returns the dummy value identity when `i > j`.
+In other words, it rotates elements in `[i, j]` one step to the right.
+-/
+def cycleIcc (i j : Fin n) : Perm (Fin n) := if hij : i ≤ j then (cycleRange ((j - i).castLT
+  (sub_val_lt_sub hij))).extendDomain (natAdd_castLEEmb (Nat.sub_le n i)).toEquivRange
+  else Equiv.refl (Fin n)
+
+theorem cycleIcc_of_lt (hij : i ≤ j) (h : k < i) : (cycleIcc i j) k = k := by
+  simpa [cycleIcc, hij] using Perm.extendDomain_apply_not_subtype _ _ (by
+    rw [range_natAdd_castLEEmb, Set.mem_setOf_eq, not_le]; omega)
+
```

Comment:
If I understand correctly, the lemma below expresses a relation between `cycleIcc` and `cycleRange`. If this is the case, it deserves a better name and not to be `private`.